airflow/providers/ssh/hooks/ssh.py

```diff
@@ -185,6 +200,18 @@ def __init__(
                     self.host_key = key_constructor(data=decoded_host_key)
                     self.no_host_key_check = False
 
+        if self.timeout:
```

Comment:
Continuing above, this check seems to make more sense if the default `timeout` is `None`.